test/e2e/checknodehealth_test.go

```diff
@@ -0,0 +1,382 @@
+				return corev1.PodUnknown
+			}
+			return podList.Items[0].Status.Phase
+		}, "20s", "5s").Should(Equal(corev1.PodPending), "Pod should remain in Pending state")
+
+		By("Waiting for pod timeout to be detected (PodTimeout = 30 seconds)")
+		var cnh *chmv1alpha1.CheckNodeHealth
+		Eventually(func() bool {
+			cnh, err = getCheckNodeHealthCR(ctx, k8sClient, cnhName)
+			if err != nil {
+				return false
+			}
+			return cnh.Status.FinishedAt != nil
+		}, "60s", "5s").Should(BeTrue(), "Pod timeout was not detected within 1 minutes")
+
+		By("Verifying timeout condition is set correctly")
+		Expect(cnh.Status.Conditions).To(HaveLen(1))
+		Expect(cnh.Status.Conditions[0].Type).To(Equal("Healthy"))
+		Expect(cnh.Status.Conditions[0].Status).To(Equal(metav1.ConditionFalse))
+
+		By("Verifying PodStartup result is recorded as Unhealthy")
+		var podStartupResult *chmv1alpha1.CheckResult
+		for i := range cnh.Status.Results {
+			if cnh.Status.Results[i].Name == "PodStartup" {
+				podStartupResult = &cnh.Status.Results[i]
+				break
+			}
+		}
+		Expect(podStartupResult).NotTo(BeNil(), "PodStartup result should be present")
+		Expect(podStartupResult.Status).To(Equal(chmv1alpha1.CheckStatusUnhealthy))
+		Expect(podStartupResult.Message).To(ContainSubstring("timeout"))
+
+		By("Verifying StartedAt timestamp is set")
+		Expect(cnh.Status.StartedAt).NotTo(BeNil())
+
+		By("Verifying the timed out pod is cleaned up")
+		Eventually(func() int {
+			podList, err := clientset.CoreV1().Pods(checkerNamespace).List(ctx, metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", checknodehealth.CheckNodeHealthLabel, cnhName),
+			})
+			if err != nil {
+				GinkgoWriter.Printf("Failed to list pods: %v\n", err)
+				return -1
+			}
+			return len(podList.Items)
+		}, "30s", "2s").Should(Equal(0), "Timed out pod was not cleaned up within timeout")
+	})
+
+	It("should cleanup pod when CR is deleted", func() {
+		By("Creating a CheckNodeHealth CR with non-existent node")
+		cnhName = fmt.Sprintf("test-cnh-deletion-%d", time.Now().Unix())
+		nonExistentNode := "fake-node-for-deletion-test"
+		err := createCheckNodeHealthCR(ctx, k8sClient, cnhName, nonExistentNode)
+		Expect(err).NotTo(HaveOccurred())
+		GinkgoWriter.Printf("Created CheckNodeHealth CR: %s with non-existent node\n", cnhName)
+
+		By("Waiting for health check pod to be created and stuck in Pending")
+		Eventually(func() bool {
+			podList, err := clientset.CoreV1().Pods(checkerNamespace).List(ctx, metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", checknodehealth.CheckNodeHealthLabel, cnhName),
+			})
+			if err != nil || len(podList.Items) == 0 {
+				return false
+			}
+			pod := &podList.Items[0]
+			return pod.Spec.NodeName == nonExistentNode && pod.Status.Phase == corev1.PodPending
+		}, "10s", "1s").Should(BeTrue(), "Health check pod was not created or not in Pending state")
+
+		By("Deleting the CheckNodeHealth CR before timeout (within 5 seconds of creation)")
```

Comment:
I don't see anything enforcing the "within 5 seconds of creation". It's also not clear to me why 5 seconds is the cutoff.